code/datums/components/defibrillator.dm

```diff
@@ -0,0 +1,382 @@
+
+	if(!should_cause_harm && (application_result & COMPONENT_DEFIB_BECOME_THE_DANGER))
+		should_cause_harm = TRUE
+	if(should_cause_harm && (application_result & COMPONENT_DEFIB_BECOME_SAFE))
+		should_cause_harm = FALSE
+
+	if(application_result & COMPONENT_BLOCK_DEFIB_DEAD)
+		user.visible_message("<span class='notice'>[defib_ref] beeps: Unit is unpowered.</span>")
+		playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+		return
+
+	if(on_cooldown)
+		to_chat(user, "<span class='notice'>[defib_ref] is recharging.</span>")
+		return
+
+	if(application_result & COMPONENT_BLOCK_DEFIB_MISC)
+		return  // the unit should handle this
+
+	if(!istype(target))
+		if(robotic)
+			to_chat(user, "<span class='notice'>This unit is only designed to work on humanoid lifeforms.</span>")
+		else
+			to_chat(user, "<span class='notice'>The instructions on [defib_ref] don't mention how to defibrillate that...</span>")
+		return
+
+	if(should_cause_harm)
+		fibrillate(user, target)
+		return
+
+	user.visible_message(
+		"<span class='warning'>[user] begins to place [parent] on [target]'s chest.</span>",
+		"<span class='warning'>You begin to place [parent] on [target.name]'s chest.</span>"
+	)
+
+	busy = TRUE
+	var/mob/dead/observer/ghost = target.get_ghost(TRUE)
+	if(ghost && ghost.can_reenter_corpse)
+		to_chat(ghost, "<span class='ghostalert'>Your heart is being defibrillated. Return to your body if you want to be revived!</span> (Verbs -> Ghost -> Re-enter corpse)")
+		window_flash(ghost.client)
+		notify_ghosts()
+		ghost << sound('sound/effects/genetics.ogg')
+
+	if(!do_after(user, 3 SECONDS * speed_multiplier, target = target)) //beginning to place the paddles on patient's chest to allow some time for people to move away to stop the process
+		busy = FALSE
+		SEND_SIGNAL(parent, COMSIG_DEFIB_ABORTED, user, target, should_cause_harm)
+		return
+
+	user.visible_message("<span class='notice'>[user] places [parent] on [target]'s chest.</span>", "<span class='warning'>You place [parent] on [target]'s chest.</span>")
+	playsound(get_turf(defib_ref), 'sound/machines/defib_charge.ogg', 50, 0)
+
+	if(ghost && !ghost.client && !QDELETED(ghost))
+		log_debug("Ghost of name [ghost.name] is bound to [target.real_name], but lacks a client. Deleting ghost.")
+		QDEL_NULL(ghost)
+
+	var/tplus = world.time - target.timeofdeath
+	var/tlimit = DEFIB_TIME_LIMIT
+	var/tloss = DEFIB_TIME_LOSS
+	var/total_burn	= 0
+	var/total_brute	= 0
+
+	var/signal_result = SEND_SIGNAL(target, COMSIG_LIVING_PRE_DEFIB, user, parent, ghost)
+
+	if(!do_after(user, 2 SECONDS * speed_multiplier, target = target)) //placed on chest and short delay to shock for dramatic effect, revive time is 5sec total
+		busy = FALSE
+		SEND_SIGNAL(parent, COMSIG_DEFIB_ABORTED, user, target, should_cause_harm)
+		return
+
+	signal_result |= SEND_SIGNAL(target, COMSIG_LIVING_DEFIBBED, user, parent, ghost)
+	if(istype(target.wear_suit, /obj/item/clothing/suit/space))
+		if(!combat)
```

Comment:
instead of adding in an if check here, just put this as the first check in the previous `if()` so you can early skip the istype and unindent the below code